packages/experience/src/hooks/use-global-redirect-to.ts

```diff
@@ -1,27 +1,47 @@
+import { noop } from '@react-spring/shared';
 import { useCallback, useContext } from 'react';
 
 import PageContext from '@/Providers/PageContextProvider/PageContext';
 import UserInteractionContext from '@/Providers/UserInteractionContextProvider/UserInteractionContext';
 
 /**
- * This hook provides a function that process the app redirection after user successfully signs in.
- * Use window.location.replace to handle the redirection.
- * Set the global loading state to true before redirecting.
- * This is to prevent the user from interacting with the app while the redirection is in progress.
+ * Hook for global redirection after successful login.
+ *
+ * This hook provides an async function that represents the final step in the login process.
+ * It sets a global loading state, clears the interaction context, and then redirects the user.
+ *
+ * The returned async function will never resolve, as the page will be redirected before
+ * the Promise can settle. This behavior is intentional and serves to maintain the loading
+ * state on the interaction element (e.g., a button) that triggered the successful login.
  */
 function useGlobalRedirectTo() {
   const { setLoading } = useContext(PageContext);
   const { clearInteractionContextSessionStorage } = useContext(UserInteractionContext);
 
   const redirectTo = useCallback(
-    (url: string | URL) => {
+    async (url: string | URL): Promise<never> => {
+      /**
+       * Set global loading state to true
+       * This prevents further user interaction during the redirect process
+       */
       setLoading(true);
       /**
        * Clear all identifier input values from the storage once the interaction is submitted.
        * The Identifier cache should be session-isolated, so it should be cleared after the interaction is completed.
        */
       clearInteractionContextSessionStorage();
+      /**
+       * Perform the actual redirect
+       * This is a synchronous operation and will immediately unload the current page
+       */
       window.location.replace(url);
+
+      /**
+       * Return a Promise that never resolves
+       * This ensures that any async function awaiting this redirect will never continue
+       * Thus maintaining the loading state on the UI until the new page is loaded
+       */
+      return new Promise<never>(noop);
```

Comment:
Seems a bit odd to me.

When do you call `setLoading(false)`?

Reply:
Seem like can be removed

Reply:
Like this unresolved promise, we don’t need to set `setIsLoading(false)` because the page will soon redirect to the login success page, and the experience page will be destroyed.

Reply:
@simeng-li 
> Seem like can be removed

It cannot be removed because setIsLoading(true) will display a global invisible LoadingMask, preventing the user from clicking on any elements on the page during the redirect loading